File: app/src/main/java/com/angogasapps/myfamily/ui/screens/settings/SettingsActivity.java
package com.angogasapps.myfamily.ui.screens.settings;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;
import android.os.Bundle;
import android.preference.PreferenceActivity;

import androidx.preference.CheckBoxPreference;
import androidx.preference.PreferenceFragmentCompat;

import com.angogasapps.myfamily.R;
import com.angogasapps.myfamily.app.AppApplication;
import com.angogasapps.myfamily.async.notification.FcmMessageManager;
import com.angogasapps.myfamily.databinding.ActivitySettingsBinding;

import es.dmoral.toasty.Toasty;


public class SettingsActivity extends AppCompatActivity {
    private ActivitySettingsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySettingsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        getSupportFragmentManager().beginTransaction()
                .add(binding.settingsContainer.getId(), new SettingFragment())
                .commit();
    }


    public static class SettingFragment extends PreferenceFragmentCompat{
        private CheckBoxPreference chatNotificationPref;
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            addPreferencesFromResource(R.xml.settings);
            chatNotificationPref = findPreference("chat_notification");

            chatNotificationPref.setOnPreferenceClickListener(preference -> {
                FcmMessageManager.INSTANCE.setPermissionToGetChatNotifications(chatNotificationPref.isChecked());
                return false;
            });
        }
    }


}

File: app/src/main/java/com/angogasapps/myfamily/ui/screens/settings/SettingsActivity.kt
package com.angogasapps.myfamily.ui.screens.settings

import com.angogasapps.myfamily.async.notification.FcmMessageManager.setPermissionToGetChatNotifications
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.preference.CheckBoxPreference
import androidx.preference.Preference
import com.angogasapps.myfamily.ui.screens.settings.SettingsActivity.SettingFragment
import androidx.preference.PreferenceFragmentCompat
import com.angogasapps.myfamily.R
import com.angogasapps.myfamily.async.notification.FcmMessageManager
import com.angogasapps.myfamily.databinding.ActivitySettingsBinding

class SettingsActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportFragmentManager.beginTransaction()
            .add(binding.settingsContainer.id, SettingFragment())
            .commit()
    }

    class SettingFragment : PreferenceFragmentCompat() {
        private lateinit var chatNotificationPref: CheckBoxPreference

        override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
            addPreferencesFromResource(R.xml.settings)
            chatNotificationPref = findPreference("chat_notification")!!
            chatNotificationPref.onPreferenceClickListener = Preference.OnPreferenceClickListener {
                setPermissionToGetChatNotifications(chatNotificationPref.isChecked)
                false
            }
        }
    }
}